perfschema/statement.go

```diff
@@ -160,7 +160,10 @@ func (ps *perfSchema) StartStatement(sql string, connID uint64, callerName EnumC
 			return nil
 		}
 		source = fmt.Sprintf("%s:%d", fileName, fileLine)
+
+		callerLock.Lock()
```

Comment:
I had second thoughts again.

- there is only one write op for the same key, so we could read map safely.
- lock out side of `if` could cause performance issue when large number of request comes concurrently.

So I think there is no need to change the code.